src/objects.rs

```diff
@@ -691,6 +691,44 @@ impl GetObjectType for VehicleJourney {
     }
 }
 
+#[derive(Debug)]
+pub enum StopTimeError {
+    DuplicateStopSequence { vj_id: String },
+    IncoherentStopTimes { vj_id: String },
+}
+
+impl std::fmt::Display for StopTimeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            StopTimeError::DuplicateStopSequence { vj_id } => {
+                write!(f, "duplicate stop_sequence for the trip '{}'", vj_id)
+            }
+            StopTimeError::IncoherentStopTimes { vj_id } => {
+                write!(f, "incoherent stop times for the trip '{}'", vj_id)
+            }
+        }
+    }
+}
+
+impl VehicleJourney {
+    pub fn sort_and_check_stop_times(&mut self) -> Result<(), StopTimeError> {
+        self.stop_times.sort_unstable_by_key(|st| st.sequence);
+        for (curr_st, next_st) in self.stop_times.iter().zip(self.stop_times.iter().skip(1)) {
+            if curr_st.sequence >= next_st.sequence {
+                return Err(StopTimeError::DuplicateStopSequence {
+                    vj_id: self.id.clone(),
+                });
+            }
+            if curr_st.departure_time > next_st.arrival_time {
```

Comment:
Shouldn't this be a "greater or equal" comparison? Because if 2 subsequent stops are delivered at the exact same time, it seems weird that it would be a valid use case... unless some data producers are not able to provide precision to the second :smile:

Reply:
some data producers are provide precision to the millisecond https://github.com/CanalTP/transit_model/pull/755

Reply:
OK, I'm resolving this comment and also [this one](https://github.com/CanalTP/transit_model/pull/768#discussion_r618981702).